knative-operator/pkg/controller/dashboard/dashboard.go

```diff
@@ -17,44 +18,38 @@ import (
 var log = common.Log.WithName("dashboard")
 
 const ConfigManagedNamespace = "openshift-config-managed"
-
-const EventingResourceDashboardPathEnvVar = "EVENTING_RESOURCES_DASHBOARD_MANIFEST_PATH"
-const EventingBrokerDashboardPathEnvVar = "EVENTING_BROKER_DASHBOARD_MANIFEST_PATH"
-const EventingSourceDashboardPathEnvVar = "EVENTING_SOURCE_DASHBOARD_MANIFEST_PATH"
-const EventingChannelDashboardPathEnvVar = "EVENTING_CHANNEL_DASHBOARD_MANIFEST_PATH"
-const ServingResourceDashboardPathEnvVar = "SERVING_RESOURCES_DASHBOARD_MANIFEST_PATH"
+const DashboardsManifestPathEnvVar = "DASHBOARDS_ROOT_MANIFEST_PATH"
 
 // Apply applies dashboard resources.
-func Apply(path string, instance operatorv1alpha1.KComponent, api client.Client) error {
+func Apply(manifestSubPath string, instance operatorv1alpha1.KComponent, api client.Client) error {
```

Comment:
It's might be a matter of preference but I like less arguments, so I would remove `manifestSubPath string` and get the subpath from instance like:

```go
func subpath(instance operatorv1alpha1.KComponent) (string, err) {
        switch instance.(type) {
        case *operatorv1alpha1.KnativeServing:
                return "serving", nil
               ...
```

Reply:
@nak3 yeah thought of that but also we had a long discussion in the past with @markusthoemmes about switching on this type. If you both agree I am ok.

Reply:
Oh, you already had a long discussion and switching on this (current PR) type. Then, I'm fine for the current PR's type.

Reply:
Not current PR but in other places at the code base where we are more in favor of being straightforward via arguments passing. At least that was my understanding.

Reply:
Well we had that discussion in the past. In this case, it's fine either way IMO but in general, yes, I personally prefer to not specialize code that seems generic from the outside API.